src/error.rs

```diff
@@ -0,0 +1,130 @@
+use std::{fmt, error};
```

Comment:
Always use `core::fmt` instead of `std::fmt`.

We cannot have a hard dependency on `std` in this crate. Please add an off-by-default "std" feature that enables the `std::error` implementation.